File: backend/app/schemas/security.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class SessionOut(BaseModel):
    id: uuid.UUID
    device_name: str | None
    user_agent: str | None
    ip_address: str | None
    is_active: bool
    last_active_at: datetime | None
    created_at: datetime
    is_current: bool = False

    model_config = {"from_attributes": True}


class LoginHistoryOut(BaseModel):
    id: uuid.UUID
    ip_address: str | None
    user_agent: str | None
    success: bool
    failure_reason: str | None
    is_suspicious: bool
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend-next/src/types/security.ts
export interface Session {
  id: string
  device_name: string | null
  user_agent: string | null
  ip_address: string | null
  is_active: boolean
  last_active_at: string | null
  created_at: string
  is_current: boolean
}

export interface LoginHistoryEntry {
  id: string
  ip_address: string | null
  user_agent: string | null
  success: boolean
  failure_reason: string | null
  is_suspicious: boolean
  created_at: string
}
